arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -1008,10 +1008,6 @@ futures::Future<Result> RocksDBMetaCollection::rebuildRevisionTree() {
     while (true) {
       std::unique_lock<std::mutex> guard(_revisionTreeLock);
 
-      if (_revisionTreeApplied < beginSeq) {
-        _revisionTreeApplied = beginSeq;
-      }
```

Comment:
This code could potentially move `_revisionTreeApplied` past a blocker. Even worse, if the timeout of two minutes is hit, we leave the old merkel tree in an invalid state.